// @testmo/testmo-api/dist/model/AutomationSource.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The AutomationSource model module.
 * @module model/AutomationSource
 * @version 2.0.0
 */
var AutomationSource = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AutomationSource</code>.
   * @alias module:model/AutomationSource
   */
  function AutomationSource() {
    _classCallCheck(this, AutomationSource);
    AutomationSource.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AutomationSource, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AutomationSource</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AutomationSource} obj Optional instance to populate.
     * @return {module:model/AutomationSource} The populated <code>AutomationSource</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AutomationSource();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('project_id')) {
          obj['project_id'] = _ApiClient["default"].convertToType(data['project_id'], 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'Number');
        }
        if (data.hasOwnProperty('is_retired')) {
          obj['is_retired'] = _ApiClient["default"].convertToType(data['is_retired'], 'Boolean');
        }
        if (data.hasOwnProperty('run_count')) {
          obj['run_count'] = _ApiClient["default"].convertToType(data['run_count'], 'Number');
        }
        if (data.hasOwnProperty('test_count_average')) {
          obj['test_count_average'] = _ApiClient["default"].convertToType(data['test_count_average'], 'Number');
        }
        if (data.hasOwnProperty('elapsed_average')) {
          obj['elapsed_average'] = _ApiClient["default"].convertToType(data['elapsed_average'], 'Number');
        }
        if (data.hasOwnProperty('ran_at')) {
          obj['ran_at'] = _ApiClient["default"].convertToType(data['ran_at'], 'Date');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('created_by')) {
          obj['created_by'] = _ApiClient["default"].convertToType(data['created_by'], 'Number');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_by')) {
          obj['updated_by'] = _ApiClient["default"].convertToType(data['updated_by'], 'Number');
        }
        if (data.hasOwnProperty('retired_at')) {
          obj['retired_at'] = _ApiClient["default"].convertToType(data['retired_at'], 'Date');
        }
        if (data.hasOwnProperty('retired_by')) {
          obj['retired_by'] = _ApiClient["default"].convertToType(data['retired_by'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>AutomationSource</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AutomationSource</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      return true;
    }
  }]);
  return AutomationSource;
}();
/**
 * @member {Number} id
 */
AutomationSource.prototype['id'] = undefined;

/**
 * @member {Number} project_id
 */
AutomationSource.prototype['project_id'] = undefined;

/**
 * @member {String} name
 */
AutomationSource.prototype['name'] = undefined;

/**
 * @member {module:model/AutomationSource.StatusEnum} status
 */
AutomationSource.prototype['status'] = undefined;

/**
 * @member {Boolean} is_retired
 */
AutomationSource.prototype['is_retired'] = undefined;

/**
 * @member {Number} run_count
 */
AutomationSource.prototype['run_count'] = undefined;

/**
 * @member {Number} test_count_average
 */
AutomationSource.prototype['test_count_average'] = undefined;

/**
 * @member {Number} elapsed_average
 */
AutomationSource.prototype['elapsed_average'] = undefined;

/**
 * @member {Date} ran_at
 */
AutomationSource.prototype['ran_at'] = undefined;

/**
 * @member {Date} created_at
 */
AutomationSource.prototype['created_at'] = undefined;

/**
 * @member {Number} created_by
 */
AutomationSource.prototype['created_by'] = undefined;

/**
 * @member {Date} updated_at
 */
AutomationSource.prototype['updated_at'] = undefined;

/**
 * @member {Number} updated_by
 */
AutomationSource.prototype['updated_by'] = undefined;

/**
 * @member {Date} retired_at
 */
AutomationSource.prototype['retired_at'] = undefined;

/**
 * @member {Number} retired_by
 */
AutomationSource.prototype['retired_by'] = undefined;

/**
 * Allowed values for the <code>status</code> property.
 * @enum {Number}
 * @readonly
 */
AutomationSource['StatusEnum'] = {
  /**
   * value: 1
   * @const
   */
  "1": 1,
  /**
   * value: 2
   * @const
   */
  "2": 2,
  /**
   * value: 3
   * @const
   */
  "3": 3
};
var _default = AutomationSource;
exports["default"] = _default;
