// @testmo/testmo-api/dist/model/AutomationRunThread.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AutomationArtifact = _interopRequireDefault(require("./AutomationArtifact"));
var _AutomationField = _interopRequireDefault(require("./AutomationField"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The AutomationRunThread model module.
 * @module model/AutomationRunThread
 * @version 2.0.0
 */
var AutomationRunThread = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AutomationRunThread</code>.
   * @alias module:model/AutomationRunThread
   */
  function AutomationRunThread() {
    _classCallCheck(this, AutomationRunThread);
    AutomationRunThread.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AutomationRunThread, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AutomationRunThread</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AutomationRunThread} obj Optional instance to populate.
     * @return {module:model/AutomationRunThread} The populated <code>AutomationRunThread</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AutomationRunThread();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('index')) {
          obj['index'] = _ApiClient["default"].convertToType(data['index'], 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'Number');
        }
        if (data.hasOwnProperty('elapsed')) {
          obj['elapsed'] = _ApiClient["default"].convertToType(data['elapsed'], 'Number');
        }
        if (data.hasOwnProperty('is_completed')) {
          obj['is_completed'] = _ApiClient["default"].convertToType(data['is_completed'], 'Boolean');
        }
        if (data.hasOwnProperty('artifacts')) {
          obj['artifacts'] = _ApiClient["default"].convertToType(data['artifacts'], [_AutomationArtifact["default"]]);
        }
        if (data.hasOwnProperty('fields')) {
          obj['fields'] = _ApiClient["default"].convertToType(data['fields'], [_AutomationField["default"]]);
        }
        if (data.hasOwnProperty('untested_count')) {
          obj['untested_count'] = _ApiClient["default"].convertToType(data['untested_count'], 'Number');
        }
        if (data.hasOwnProperty('status1_count')) {
          obj['status1_count'] = _ApiClient["default"].convertToType(data['status1_count'], 'Number');
        }
        if (data.hasOwnProperty('status2_count')) {
          obj['status2_count'] = _ApiClient["default"].convertToType(data['status2_count'], 'Number');
        }
        if (data.hasOwnProperty('status3_count')) {
          obj['status3_count'] = _ApiClient["default"].convertToType(data['status3_count'], 'Number');
        }
        if (data.hasOwnProperty('status4_count')) {
          obj['status4_count'] = _ApiClient["default"].convertToType(data['status4_count'], 'Number');
        }
        if (data.hasOwnProperty('status5_count')) {
          obj['status5_count'] = _ApiClient["default"].convertToType(data['status5_count'], 'Number');
        }
        if (data.hasOwnProperty('status6_count')) {
          obj['status6_count'] = _ApiClient["default"].convertToType(data['status6_count'], 'Number');
        }
        if (data.hasOwnProperty('status7_count')) {
          obj['status7_count'] = _ApiClient["default"].convertToType(data['status7_count'], 'Number');
        }
        if (data.hasOwnProperty('status8_count')) {
          obj['status8_count'] = _ApiClient["default"].convertToType(data['status8_count'], 'Number');
        }
        if (data.hasOwnProperty('status9_count')) {
          obj['status9_count'] = _ApiClient["default"].convertToType(data['status9_count'], 'Number');
        }
        if (data.hasOwnProperty('status10_count')) {
          obj['status10_count'] = _ApiClient["default"].convertToType(data['status10_count'], 'Number');
        }
        if (data.hasOwnProperty('status11_count')) {
          obj['status11_count'] = _ApiClient["default"].convertToType(data['status11_count'], 'Number');
        }
        if (data.hasOwnProperty('status12_count')) {
          obj['status12_count'] = _ApiClient["default"].convertToType(data['status12_count'], 'Number');
        }
        if (data.hasOwnProperty('status13_count')) {
          obj['status13_count'] = _ApiClient["default"].convertToType(data['status13_count'], 'Number');
        }
        if (data.hasOwnProperty('status14_count')) {
          obj['status14_count'] = _ApiClient["default"].convertToType(data['status14_count'], 'Number');
        }
        if (data.hasOwnProperty('status15_count')) {
          obj['status15_count'] = _ApiClient["default"].convertToType(data['status15_count'], 'Number');
        }
        if (data.hasOwnProperty('status16_count')) {
          obj['status16_count'] = _ApiClient["default"].convertToType(data['status16_count'], 'Number');
        }
        if (data.hasOwnProperty('status17_count')) {
          obj['status17_count'] = _ApiClient["default"].convertToType(data['status17_count'], 'Number');
        }
        if (data.hasOwnProperty('status18_count')) {
          obj['status18_count'] = _ApiClient["default"].convertToType(data['status18_count'], 'Number');
        }
        if (data.hasOwnProperty('status19_count')) {
          obj['status19_count'] = _ApiClient["default"].convertToType(data['status19_count'], 'Number');
        }
        if (data.hasOwnProperty('status20_count')) {
          obj['status20_count'] = _ApiClient["default"].convertToType(data['status20_count'], 'Number');
        }
        if (data.hasOwnProperty('status21_count')) {
          obj['status21_count'] = _ApiClient["default"].convertToType(data['status21_count'], 'Number');
        }
        if (data.hasOwnProperty('status22_count')) {
          obj['status22_count'] = _ApiClient["default"].convertToType(data['status22_count'], 'Number');
        }
        if (data.hasOwnProperty('status23_count')) {
          obj['status23_count'] = _ApiClient["default"].convertToType(data['status23_count'], 'Number');
        }
        if (data.hasOwnProperty('status24_count')) {
          obj['status24_count'] = _ApiClient["default"].convertToType(data['status24_count'], 'Number');
        }
        if (data.hasOwnProperty('success_count')) {
          obj['success_count'] = _ApiClient["default"].convertToType(data['success_count'], 'Number');
        }
        if (data.hasOwnProperty('failure_count')) {
          obj['failure_count'] = _ApiClient["default"].convertToType(data['failure_count'], 'Number');
        }
        if (data.hasOwnProperty('completed_count')) {
          obj['completed_count'] = _ApiClient["default"].convertToType(data['completed_count'], 'Number');
        }
        if (data.hasOwnProperty('total_count')) {
          obj['total_count'] = _ApiClient["default"].convertToType(data['total_count'], 'Number');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('created_by')) {
          obj['created_by'] = _ApiClient["default"].convertToType(data['created_by'], 'Number');
        }
        if (data.hasOwnProperty('completed_at')) {
          obj['completed_at'] = _ApiClient["default"].convertToType(data['completed_at'], 'Date');
        }
        if (data.hasOwnProperty('completed_by')) {
          obj['completed_by'] = _ApiClient["default"].convertToType(data['completed_by'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>AutomationRunThread</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AutomationRunThread</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      if (data['artifacts']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['artifacts'])) {
          throw new Error("Expected the field `artifacts` to be an array in the JSON data but got " + data['artifacts']);
        }
        // validate the optional field `artifacts` (array)
        var _iterator = _createForOfIteratorHelper(data['artifacts']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _AutomationArtifact["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      if (data['fields']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['fields'])) {
          throw new Error("Expected the field `fields` to be an array in the JSON data but got " + data['fields']);
        }
        // validate the optional field `fields` (array)
        var _iterator2 = _createForOfIteratorHelper(data['fields']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;
            _AutomationField["default"].validateJsonObject(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      return true;
    }
  }]);
  return AutomationRunThread;
}();
/**
 * @member {Number} id
 */
AutomationRunThread.prototype['id'] = undefined;

/**
 * @member {Number} index
 */
AutomationRunThread.prototype['index'] = undefined;

/**
 * @member {String} name
 */
AutomationRunThread.prototype['name'] = undefined;

/**
 * @member {module:model/AutomationRunThread.StatusEnum} status
 */
AutomationRunThread.prototype['status'] = undefined;

/**
 * @member {Number} elapsed
 */
AutomationRunThread.prototype['elapsed'] = undefined;

/**
 * @member {Boolean} is_completed
 */
AutomationRunThread.prototype['is_completed'] = undefined;

/**
 * @member {Array.<module:model/AutomationArtifact>} artifacts
 */
AutomationRunThread.prototype['artifacts'] = undefined;

/**
 * @member {Array.<module:model/AutomationField>} fields
 */
AutomationRunThread.prototype['fields'] = undefined;

/**
 * @member {Number} untested_count
 */
AutomationRunThread.prototype['untested_count'] = undefined;

/**
 * @member {Number} status1_count
 */
AutomationRunThread.prototype['status1_count'] = undefined;

/**
 * @member {Number} status2_count
 */
AutomationRunThread.prototype['status2_count'] = undefined;

/**
 * @member {Number} status3_count
 */
AutomationRunThread.prototype['status3_count'] = undefined;

/**
 * @member {Number} status4_count
 */
AutomationRunThread.prototype['status4_count'] = undefined;

/**
 * @member {Number} status5_count
 */
AutomationRunThread.prototype['status5_count'] = undefined;

/**
 * @member {Number} status6_count
 */
AutomationRunThread.prototype['status6_count'] = undefined;

/**
 * @member {Number} status7_count
 */
AutomationRunThread.prototype['status7_count'] = undefined;

/**
 * @member {Number} status8_count
 */
AutomationRunThread.prototype['status8_count'] = undefined;

/**
 * @member {Number} status9_count
 */
AutomationRunThread.prototype['status9_count'] = undefined;

/**
 * @member {Number} status10_count
 */
AutomationRunThread.prototype['status10_count'] = undefined;

/**
 * @member {Number} status11_count
 */
AutomationRunThread.prototype['status11_count'] = undefined;

/**
 * @member {Number} status12_count
 */
AutomationRunThread.prototype['status12_count'] = undefined;

/**
 * @member {Number} status13_count
 */
AutomationRunThread.prototype['status13_count'] = undefined;

/**
 * @member {Number} status14_count
 */
AutomationRunThread.prototype['status14_count'] = undefined;

/**
 * @member {Number} status15_count
 */
AutomationRunThread.prototype['status15_count'] = undefined;

/**
 * @member {Number} status16_count
 */
AutomationRunThread.prototype['status16_count'] = undefined;

/**
 * @member {Number} status17_count
 */
AutomationRunThread.prototype['status17_count'] = undefined;

/**
 * @member {Number} status18_count
 */
AutomationRunThread.prototype['status18_count'] = undefined;

/**
 * @member {Number} status19_count
 */
AutomationRunThread.prototype['status19_count'] = undefined;

/**
 * @member {Number} status20_count
 */
AutomationRunThread.prototype['status20_count'] = undefined;

/**
 * @member {Number} status21_count
 */
AutomationRunThread.prototype['status21_count'] = undefined;

/**
 * @member {Number} status22_count
 */
AutomationRunThread.prototype['status22_count'] = undefined;

/**
 * @member {Number} status23_count
 */
AutomationRunThread.prototype['status23_count'] = undefined;

/**
 * @member {Number} status24_count
 */
AutomationRunThread.prototype['status24_count'] = undefined;

/**
 * @member {Number} success_count
 */
AutomationRunThread.prototype['success_count'] = undefined;

/**
 * @member {Number} failure_count
 */
AutomationRunThread.prototype['failure_count'] = undefined;

/**
 * @member {Number} completed_count
 */
AutomationRunThread.prototype['completed_count'] = undefined;

/**
 * @member {Number} total_count
 */
AutomationRunThread.prototype['total_count'] = undefined;

/**
 * @member {Date} created_at
 */
AutomationRunThread.prototype['created_at'] = undefined;

/**
 * @member {Number} created_by
 */
AutomationRunThread.prototype['created_by'] = undefined;

/**
 * @member {Date} completed_at
 */
AutomationRunThread.prototype['completed_at'] = undefined;

/**
 * @member {Number} completed_by
 */
AutomationRunThread.prototype['completed_by'] = undefined;

/**
 * Allowed values for the <code>status</code> property.
 * @enum {Number}
 * @readonly
 */
AutomationRunThread['StatusEnum'] = {
  /**
   * value: 2
   * @const
   */
  "2": 2,
  /**
   * value: 3
   * @const
   */
  "3": 3,
  /**
   * value: 4
   * @const
   */
  "4": 4
};
var _default = AutomationRunThread;
exports["default"] = _default;
