// @testmo/testmo-api/dist/model/AutomationRunRow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AutomationRun = _interopRequireDefault(require("./AutomationRun"));
var _AutomationRunExpands = _interopRequireDefault(require("./AutomationRunExpands"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The AutomationRunRow model module.
 * @module model/AutomationRunRow
 * @version 2.0.0
 */
var AutomationRunRow = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AutomationRunRow</code>.
   * @alias module:model/AutomationRunRow
   */
  function AutomationRunRow() {
    _classCallCheck(this, AutomationRunRow);
    AutomationRunRow.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AutomationRunRow, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AutomationRunRow</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AutomationRunRow} obj Optional instance to populate.
     * @return {module:model/AutomationRunRow} The populated <code>AutomationRunRow</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AutomationRunRow();
        if (data.hasOwnProperty('result')) {
          obj['result'] = _AutomationRun["default"].constructFromObject(data['result']);
        }
        if (data.hasOwnProperty('expands')) {
          obj['expands'] = _AutomationRunExpands["default"].constructFromObject(data['expands']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>AutomationRunRow</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AutomationRunRow</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // validate the optional field `result`
      if (data['result']) {
        // data not null
        _AutomationRun["default"].validateJSON(data['result']);
      }
      // validate the optional field `expands`
      if (data['expands']) {
        // data not null
        _AutomationRunExpands["default"].validateJSON(data['expands']);
      }
      return true;
    }
  }]);
  return AutomationRunRow;
}();
/**
 * @member {module:model/AutomationRun} result
 */
AutomationRunRow.prototype['result'] = undefined;

/**
 * @member {module:model/AutomationRunExpands} expands
 */
AutomationRunRow.prototype['expands'] = undefined;
var _default = AutomationRunRow;
exports["default"] = _default;
