// @testmo/testmo-api/dist/model/AutomationSourcePage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AutomationSource = _interopRequireDefault(require("./AutomationSource"));
var _AutomationSourceExpands = _interopRequireDefault(require("./AutomationSourceExpands"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The AutomationSourcePage model module.
 * @module model/AutomationSourcePage
 * @version 2.0.0
 */
var AutomationSourcePage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AutomationSourcePage</code>.
   * @alias module:model/AutomationSourcePage
   */
  function AutomationSourcePage() {
    _classCallCheck(this, AutomationSourcePage);
    AutomationSourcePage.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AutomationSourcePage, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AutomationSourcePage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AutomationSourcePage} obj Optional instance to populate.
     * @return {module:model/AutomationSourcePage} The populated <code>AutomationSourcePage</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AutomationSourcePage();
        if (data.hasOwnProperty('page')) {
          obj['page'] = _ApiClient["default"].convertToType(data['page'], 'Number');
        }
        if (data.hasOwnProperty('prev_page')) {
          obj['prev_page'] = _ApiClient["default"].convertToType(data['prev_page'], 'Number');
        }
        if (data.hasOwnProperty('next_page')) {
          obj['next_page'] = _ApiClient["default"].convertToType(data['next_page'], 'Number');
        }
        if (data.hasOwnProperty('last_page')) {
          obj['last_page'] = _ApiClient["default"].convertToType(data['last_page'], 'Number');
        }
        if (data.hasOwnProperty('per_page')) {
          obj['per_page'] = _ApiClient["default"].convertToType(data['per_page'], 'Number');
        }
        if (data.hasOwnProperty('total')) {
          obj['total'] = _ApiClient["default"].convertToType(data['total'], 'Number');
        }
        if (data.hasOwnProperty('result')) {
          obj['result'] = _ApiClient["default"].convertToType(data['result'], [_AutomationSource["default"]]);
        }
        if (data.hasOwnProperty('expands')) {
          obj['expands'] = _AutomationSourceExpands["default"].constructFromObject(data['expands']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>AutomationSourcePage</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AutomationSourcePage</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['result']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['result'])) {
          throw new Error("Expected the field `result` to be an array in the JSON data but got " + data['result']);
        }
        // validate the optional field `result` (array)
        var _iterator = _createForOfIteratorHelper(data['result']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _AutomationSource["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      // validate the optional field `expands`
      if (data['expands']) {
        // data not null
        _AutomationSourceExpands["default"].validateJSON(data['expands']);
      }
      return true;
    }
  }]);
  return AutomationSourcePage;
}();
/**
 * @member {Number} page
 */
AutomationSourcePage.prototype['page'] = undefined;

/**
 * @member {Number} prev_page
 */
AutomationSourcePage.prototype['prev_page'] = undefined;

/**
 * @member {Number} next_page
 */
AutomationSourcePage.prototype['next_page'] = undefined;

/**
 * @member {Number} last_page
 */
AutomationSourcePage.prototype['last_page'] = undefined;

/**
 * @member {Number} per_page
 */
AutomationSourcePage.prototype['per_page'] = undefined;

/**
 * @member {Number} total
 */
AutomationSourcePage.prototype['total'] = undefined;

/**
 * @member {Array.<module:model/AutomationSource>} result
 */
AutomationSourcePage.prototype['result'] = undefined;

/**
 * @member {module:model/AutomationSourceExpands} expands
 */
AutomationSourcePage.prototype['expands'] = undefined;
var _default = AutomationSourcePage;
exports["default"] = _default;
