// @testmo/testmo-api/dist/model/CreateAutomationRunThread.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CreateAutomationArtifact = _interopRequireDefault(require("./CreateAutomationArtifact"));
var _CreateAutomationField = _interopRequireDefault(require("./CreateAutomationField"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The CreateAutomationRunThread model module.
 * @module model/CreateAutomationRunThread
 * @version 2.0.0
 */
var CreateAutomationRunThread = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAutomationRunThread</code>.
   * @alias module:model/CreateAutomationRunThread
   */
  function CreateAutomationRunThread() {
    _classCallCheck(this, CreateAutomationRunThread);
    CreateAutomationRunThread.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(CreateAutomationRunThread, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>CreateAutomationRunThread</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAutomationRunThread} obj Optional instance to populate.
     * @return {module:model/CreateAutomationRunThread} The populated <code>CreateAutomationRunThread</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAutomationRunThread();
        if (data.hasOwnProperty('elapsed_observed')) {
          obj['elapsed_observed'] = _ApiClient["default"].convertToType(data['elapsed_observed'], 'Number');
        }
        if (data.hasOwnProperty('elapsed_computed')) {
          obj['elapsed_computed'] = _ApiClient["default"].convertToType(data['elapsed_computed'], 'Number');
        }
        if (data.hasOwnProperty('artifacts')) {
          obj['artifacts'] = _ApiClient["default"].convertToType(data['artifacts'], [_CreateAutomationArtifact["default"]]);
        }
        if (data.hasOwnProperty('fields')) {
          obj['fields'] = _ApiClient["default"].convertToType(data['fields'], [_CreateAutomationField["default"]]);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>CreateAutomationRunThread</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>CreateAutomationRunThread</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['artifacts']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['artifacts'])) {
          throw new Error("Expected the field `artifacts` to be an array in the JSON data but got " + data['artifacts']);
        }
        // validate the optional field `artifacts` (array)
        var _iterator = _createForOfIteratorHelper(data['artifacts']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _CreateAutomationArtifact["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      if (data['fields']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['fields'])) {
          throw new Error("Expected the field `fields` to be an array in the JSON data but got " + data['fields']);
        }
        // validate the optional field `fields` (array)
        var _iterator2 = _createForOfIteratorHelper(data['fields']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;
            _CreateAutomationField["default"].validateJsonObject(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      return true;
    }
  }]);
  return CreateAutomationRunThread;
}();
/**
 * Observed overall elapsed (execution time) of the thread in microseconds.
 * @member {Number} elapsed_observed
 */
CreateAutomationRunThread.prototype['elapsed_observed'] = undefined;

/**
 * Computed overall elapsed (execution time) of the thread in microseconds. The _observed_ elapsed is usually derived from measuring the execution time of tests whereas the _computed_ elapsed is calculated from result log files.
 * @member {Number} elapsed_computed
 */
CreateAutomationRunThread.prototype['elapsed_computed'] = undefined;

/**
 * List of externally stored test artifacts to link to the new thread (such as log files, screenshots or test data).
 * @member {Array.<module:model/CreateAutomationArtifact>} artifacts
 */
CreateAutomationRunThread.prototype['artifacts'] = undefined;

/**
 * List of fields to attach to the new thread (such as environment variables, errors or terminal output).
 * @member {Array.<module:model/CreateAutomationField>} fields
 */
CreateAutomationRunThread.prototype['fields'] = undefined;
var _default = CreateAutomationRunThread;
exports["default"] = _default;
