// @testmo/testmo-api/dist/model/IssueExpand.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The IssueExpand model module.
 * @module model/IssueExpand
 * @version 2.0.0
 */
var IssueExpand = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IssueExpand</code>.
   * @alias module:model/IssueExpand
   */
  function IssueExpand() {
    _classCallCheck(this, IssueExpand);
    IssueExpand.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(IssueExpand, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>IssueExpand</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/IssueExpand} obj Optional instance to populate.
     * @return {module:model/IssueExpand} The populated <code>IssueExpand</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IssueExpand();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'Number');
        }
        if (data.hasOwnProperty('display_id')) {
          obj['display_id'] = _ApiClient["default"].convertToType(data['display_id'], 'String');
        }
        if (data.hasOwnProperty('external_project_id')) {
          obj['external_project_id'] = _ApiClient["default"].convertToType(data['external_project_id'], 'Number');
        }
        if (data.hasOwnProperty('external_project_key')) {
          obj['external_project_key'] = _ApiClient["default"].convertToType(data['external_project_key'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>IssueExpand</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>IssueExpand</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['display_id'] && !(typeof data['display_id'] === 'string' || data['display_id'] instanceof String)) {
        throw new Error("Expected the field `display_id` to be a primitive type in the JSON string but got " + data['display_id']);
      }
      // ensure the json data is a string
      if (data['external_project_key'] && !(typeof data['external_project_key'] === 'string' || data['external_project_key'] instanceof String)) {
        throw new Error("Expected the field `external_project_key` to be a primitive type in the JSON string but got " + data['external_project_key']);
      }
      return true;
    }
  }]);
  return IssueExpand;
}();
/**
 * @member {Number} id
 */
IssueExpand.prototype['id'] = undefined;

/**
 * @member {module:model/IssueExpand.TypeEnum} type
 */
IssueExpand.prototype['type'] = undefined;

/**
 * @member {String} display_id
 */
IssueExpand.prototype['display_id'] = undefined;

/**
 * @member {Number} external_project_id
 */
IssueExpand.prototype['external_project_id'] = undefined;

/**
 * @member {String} external_project_key
 */
IssueExpand.prototype['external_project_key'] = undefined;

/**
 * Allowed values for the <code>type</code> property.
 * @enum {Number}
 * @readonly
 */
IssueExpand['TypeEnum'] = {
  /**
   * value: 1
   * @const
   */
  "1": 1,
  /**
   * value: 2
   * @const
   */
  "2": 2,
  /**
   * value: 3
   * @const
   */
  "3": 3,
  /**
   * value: 4
   * @const
   */
  "4": 4
};
var _default = IssueExpand;
exports["default"] = _default;
