// @testmo/testmo-api/dist/model/MilestoneStatsExpand.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The MilestoneStatsExpand model module.
 * @module model/MilestoneStatsExpand
 * @version 2.0.0
 */
var MilestoneStatsExpand = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MilestoneStatsExpand</code>.
   * @alias module:model/MilestoneStatsExpand
   */
  function MilestoneStatsExpand() {
    _classCallCheck(this, MilestoneStatsExpand);
    MilestoneStatsExpand.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(MilestoneStatsExpand, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>MilestoneStatsExpand</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MilestoneStatsExpand} obj Optional instance to populate.
     * @return {module:model/MilestoneStatsExpand} The populated <code>MilestoneStatsExpand</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MilestoneStatsExpand();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('untested_count')) {
          obj['untested_count'] = _ApiClient["default"].convertToType(data['untested_count'], 'Number');
        }
        if (data.hasOwnProperty('status1_count')) {
          obj['status1_count'] = _ApiClient["default"].convertToType(data['status1_count'], 'Number');
        }
        if (data.hasOwnProperty('status2_count')) {
          obj['status2_count'] = _ApiClient["default"].convertToType(data['status2_count'], 'Number');
        }
        if (data.hasOwnProperty('status3_count')) {
          obj['status3_count'] = _ApiClient["default"].convertToType(data['status3_count'], 'Number');
        }
        if (data.hasOwnProperty('status4_count')) {
          obj['status4_count'] = _ApiClient["default"].convertToType(data['status4_count'], 'Number');
        }
        if (data.hasOwnProperty('status5_count')) {
          obj['status5_count'] = _ApiClient["default"].convertToType(data['status5_count'], 'Number');
        }
        if (data.hasOwnProperty('status6_count')) {
          obj['status6_count'] = _ApiClient["default"].convertToType(data['status6_count'], 'Number');
        }
        if (data.hasOwnProperty('status7_count')) {
          obj['status7_count'] = _ApiClient["default"].convertToType(data['status7_count'], 'Number');
        }
        if (data.hasOwnProperty('status8_count')) {
          obj['status8_count'] = _ApiClient["default"].convertToType(data['status8_count'], 'Number');
        }
        if (data.hasOwnProperty('status9_count')) {
          obj['status9_count'] = _ApiClient["default"].convertToType(data['status9_count'], 'Number');
        }
        if (data.hasOwnProperty('status10_count')) {
          obj['status10_count'] = _ApiClient["default"].convertToType(data['status10_count'], 'Number');
        }
        if (data.hasOwnProperty('status11_count')) {
          obj['status11_count'] = _ApiClient["default"].convertToType(data['status11_count'], 'Number');
        }
        if (data.hasOwnProperty('status12_count')) {
          obj['status12_count'] = _ApiClient["default"].convertToType(data['status12_count'], 'Number');
        }
        if (data.hasOwnProperty('status13_count')) {
          obj['status13_count'] = _ApiClient["default"].convertToType(data['status13_count'], 'Number');
        }
        if (data.hasOwnProperty('status14_count')) {
          obj['status14_count'] = _ApiClient["default"].convertToType(data['status14_count'], 'Number');
        }
        if (data.hasOwnProperty('status15_count')) {
          obj['status15_count'] = _ApiClient["default"].convertToType(data['status15_count'], 'Number');
        }
        if (data.hasOwnProperty('status16_count')) {
          obj['status16_count'] = _ApiClient["default"].convertToType(data['status16_count'], 'Number');
        }
        if (data.hasOwnProperty('status17_count')) {
          obj['status17_count'] = _ApiClient["default"].convertToType(data['status17_count'], 'Number');
        }
        if (data.hasOwnProperty('status18_count')) {
          obj['status18_count'] = _ApiClient["default"].convertToType(data['status18_count'], 'Number');
        }
        if (data.hasOwnProperty('status19_count')) {
          obj['status19_count'] = _ApiClient["default"].convertToType(data['status19_count'], 'Number');
        }
        if (data.hasOwnProperty('status20_count')) {
          obj['status20_count'] = _ApiClient["default"].convertToType(data['status20_count'], 'Number');
        }
        if (data.hasOwnProperty('status21_count')) {
          obj['status21_count'] = _ApiClient["default"].convertToType(data['status21_count'], 'Number');
        }
        if (data.hasOwnProperty('status22_count')) {
          obj['status22_count'] = _ApiClient["default"].convertToType(data['status22_count'], 'Number');
        }
        if (data.hasOwnProperty('status23_count')) {
          obj['status23_count'] = _ApiClient["default"].convertToType(data['status23_count'], 'Number');
        }
        if (data.hasOwnProperty('status24_count')) {
          obj['status24_count'] = _ApiClient["default"].convertToType(data['status24_count'], 'Number');
        }
        if (data.hasOwnProperty('success_count')) {
          obj['success_count'] = _ApiClient["default"].convertToType(data['success_count'], 'Number');
        }
        if (data.hasOwnProperty('failure_count')) {
          obj['failure_count'] = _ApiClient["default"].convertToType(data['failure_count'], 'Number');
        }
        if (data.hasOwnProperty('completed_count')) {
          obj['completed_count'] = _ApiClient["default"].convertToType(data['completed_count'], 'Number');
        }
        if (data.hasOwnProperty('total_count')) {
          obj['total_count'] = _ApiClient["default"].convertToType(data['total_count'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>MilestoneStatsExpand</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>MilestoneStatsExpand</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      return true;
    }
  }]);
  return MilestoneStatsExpand;
}();
/**
 * @member {Number} id
 */
MilestoneStatsExpand.prototype['id'] = undefined;

/**
 * @member {Number} untested_count
 */
MilestoneStatsExpand.prototype['untested_count'] = undefined;

/**
 * @member {Number} status1_count
 */
MilestoneStatsExpand.prototype['status1_count'] = undefined;

/**
 * @member {Number} status2_count
 */
MilestoneStatsExpand.prototype['status2_count'] = undefined;

/**
 * @member {Number} status3_count
 */
MilestoneStatsExpand.prototype['status3_count'] = undefined;

/**
 * @member {Number} status4_count
 */
MilestoneStatsExpand.prototype['status4_count'] = undefined;

/**
 * @member {Number} status5_count
 */
MilestoneStatsExpand.prototype['status5_count'] = undefined;

/**
 * @member {Number} status6_count
 */
MilestoneStatsExpand.prototype['status6_count'] = undefined;

/**
 * @member {Number} status7_count
 */
MilestoneStatsExpand.prototype['status7_count'] = undefined;

/**
 * @member {Number} status8_count
 */
MilestoneStatsExpand.prototype['status8_count'] = undefined;

/**
 * @member {Number} status9_count
 */
MilestoneStatsExpand.prototype['status9_count'] = undefined;

/**
 * @member {Number} status10_count
 */
MilestoneStatsExpand.prototype['status10_count'] = undefined;

/**
 * @member {Number} status11_count
 */
MilestoneStatsExpand.prototype['status11_count'] = undefined;

/**
 * @member {Number} status12_count
 */
MilestoneStatsExpand.prototype['status12_count'] = undefined;

/**
 * @member {Number} status13_count
 */
MilestoneStatsExpand.prototype['status13_count'] = undefined;

/**
 * @member {Number} status14_count
 */
MilestoneStatsExpand.prototype['status14_count'] = undefined;

/**
 * @member {Number} status15_count
 */
MilestoneStatsExpand.prototype['status15_count'] = undefined;

/**
 * @member {Number} status16_count
 */
MilestoneStatsExpand.prototype['status16_count'] = undefined;

/**
 * @member {Number} status17_count
 */
MilestoneStatsExpand.prototype['status17_count'] = undefined;

/**
 * @member {Number} status18_count
 */
MilestoneStatsExpand.prototype['status18_count'] = undefined;

/**
 * @member {Number} status19_count
 */
MilestoneStatsExpand.prototype['status19_count'] = undefined;

/**
 * @member {Number} status20_count
 */
MilestoneStatsExpand.prototype['status20_count'] = undefined;

/**
 * @member {Number} status21_count
 */
MilestoneStatsExpand.prototype['status21_count'] = undefined;

/**
 * @member {Number} status22_count
 */
MilestoneStatsExpand.prototype['status22_count'] = undefined;

/**
 * @member {Number} status23_count
 */
MilestoneStatsExpand.prototype['status23_count'] = undefined;

/**
 * @member {Number} status24_count
 */
MilestoneStatsExpand.prototype['status24_count'] = undefined;

/**
 * @member {Number} success_count
 */
MilestoneStatsExpand.prototype['success_count'] = undefined;

/**
 * @member {Number} failure_count
 */
MilestoneStatsExpand.prototype['failure_count'] = undefined;

/**
 * @member {Number} completed_count
 */
MilestoneStatsExpand.prototype['completed_count'] = undefined;

/**
 * @member {Number} total_count
 */
MilestoneStatsExpand.prototype['total_count'] = undefined;
var _default = MilestoneStatsExpand;
exports["default"] = _default;
