// @testmo/testmo-api/dist/model/Milestone.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _MilestoneLink = _interopRequireDefault(require("./MilestoneLink"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Milestone model module.
 * @module model/Milestone
 * @version 2.0.0
 */
var Milestone = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Milestone</code>.
   * @alias module:model/Milestone
   */
  function Milestone() {
    _classCallCheck(this, Milestone);
    Milestone.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Milestone, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Milestone</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Milestone} obj Optional instance to populate.
     * @return {module:model/Milestone} The populated <code>Milestone</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Milestone();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('project_id')) {
          obj['project_id'] = _ApiClient["default"].convertToType(data['project_id'], 'Number');
        }
        if (data.hasOwnProperty('root_id')) {
          obj['root_id'] = _ApiClient["default"].convertToType(data['root_id'], 'Number');
        }
        if (data.hasOwnProperty('parent_id')) {
          obj['parent_id'] = _ApiClient["default"].convertToType(data['parent_id'], 'Number');
        }
        if (data.hasOwnProperty('type_id')) {
          obj['type_id'] = _ApiClient["default"].convertToType(data['type_id'], 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('note')) {
          obj['note'] = _ApiClient["default"].convertToType(data['note'], 'String');
        }
        if (data.hasOwnProperty('is_started')) {
          obj['is_started'] = _ApiClient["default"].convertToType(data['is_started'], 'Boolean');
        }
        if (data.hasOwnProperty('is_completed')) {
          obj['is_completed'] = _ApiClient["default"].convertToType(data['is_completed'], 'Boolean');
        }
        if (data.hasOwnProperty('start_date')) {
          obj['start_date'] = _ApiClient["default"].convertToType(data['start_date'], 'String');
        }
        if (data.hasOwnProperty('due_date')) {
          obj['due_date'] = _ApiClient["default"].convertToType(data['due_date'], 'String');
        }
        if (data.hasOwnProperty('automation_tags')) {
          obj['automation_tags'] = _ApiClient["default"].convertToType(data['automation_tags'], ['String']);
        }
        if (data.hasOwnProperty('issues')) {
          obj['issues'] = _ApiClient["default"].convertToType(data['issues'], ['Number']);
        }
        if (data.hasOwnProperty('links')) {
          obj['links'] = _ApiClient["default"].convertToType(data['links'], [_MilestoneLink["default"]]);
        }
        if (data.hasOwnProperty('started_at')) {
          obj['started_at'] = _ApiClient["default"].convertToType(data['started_at'], 'Date');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('created_by')) {
          obj['created_by'] = _ApiClient["default"].convertToType(data['created_by'], 'Number');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_by')) {
          obj['updated_by'] = _ApiClient["default"].convertToType(data['updated_by'], 'Number');
        }
        if (data.hasOwnProperty('completed_at')) {
          obj['completed_at'] = _ApiClient["default"].convertToType(data['completed_at'], 'Date');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Milestone</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Milestone</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['note'] && !(typeof data['note'] === 'string' || data['note'] instanceof String)) {
        throw new Error("Expected the field `note` to be a primitive type in the JSON string but got " + data['note']);
      }
      // ensure the json data is a string
      if (data['start_date'] && !(typeof data['start_date'] === 'string' || data['start_date'] instanceof String)) {
        throw new Error("Expected the field `start_date` to be a primitive type in the JSON string but got " + data['start_date']);
      }
      // ensure the json data is a string
      if (data['due_date'] && !(typeof data['due_date'] === 'string' || data['due_date'] instanceof String)) {
        throw new Error("Expected the field `due_date` to be a primitive type in the JSON string but got " + data['due_date']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['automation_tags'])) {
        throw new Error("Expected the field `automation_tags` to be an array in the JSON data but got " + data['automation_tags']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['issues'])) {
        throw new Error("Expected the field `issues` to be an array in the JSON data but got " + data['issues']);
      }
      if (data['links']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['links'])) {
          throw new Error("Expected the field `links` to be an array in the JSON data but got " + data['links']);
        }
        // validate the optional field `links` (array)
        var _iterator = _createForOfIteratorHelper(data['links']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _MilestoneLink["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      return true;
    }
  }]);
  return Milestone;
}();
/**
 * @member {Number} id
 */
Milestone.prototype['id'] = undefined;

/**
 * @member {Number} project_id
 */
Milestone.prototype['project_id'] = undefined;

/**
 * @member {Number} root_id
 */
Milestone.prototype['root_id'] = undefined;

/**
 * @member {Number} parent_id
 */
Milestone.prototype['parent_id'] = undefined;

/**
 * @member {Number} type_id
 */
Milestone.prototype['type_id'] = undefined;

/**
 * @member {String} name
 */
Milestone.prototype['name'] = undefined;

/**
 * @member {String} note
 */
Milestone.prototype['note'] = undefined;

/**
 * @member {Boolean} is_started
 */
Milestone.prototype['is_started'] = undefined;

/**
 * @member {Boolean} is_completed
 */
Milestone.prototype['is_completed'] = undefined;

/**
 * @member {String} start_date
 */
Milestone.prototype['start_date'] = undefined;

/**
 * @member {String} due_date
 */
Milestone.prototype['due_date'] = undefined;

/**
 * @member {Array.<String>} automation_tags
 */
Milestone.prototype['automation_tags'] = undefined;

/**
 * @member {Array.<Number>} issues
 */
Milestone.prototype['issues'] = undefined;

/**
 * @member {Array.<module:model/MilestoneLink>} links
 */
Milestone.prototype['links'] = undefined;

/**
 * @member {Date} started_at
 */
Milestone.prototype['started_at'] = undefined;

/**
 * @member {Date} created_at
 */
Milestone.prototype['created_at'] = undefined;

/**
 * @member {Number} created_by
 */
Milestone.prototype['created_by'] = undefined;

/**
 * @member {Date} updated_at
 */
Milestone.prototype['updated_at'] = undefined;

/**
 * @member {Number} updated_by
 */
Milestone.prototype['updated_by'] = undefined;

/**
 * @member {Date} completed_at
 */
Milestone.prototype['completed_at'] = undefined;
var _default = Milestone;
exports["default"] = _default;
