// @testmo/testmo-api/dist/model/MilestoneExpands.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _IssueExpand = _interopRequireDefault(require("./IssueExpand"));
var _MilestoneExpand = _interopRequireDefault(require("./MilestoneExpand"));
var _MilestoneStatsExpand = _interopRequireDefault(require("./MilestoneStatsExpand"));
var _MilestoneTypeExpand = _interopRequireDefault(require("./MilestoneTypeExpand"));
var _StatusExpand = _interopRequireDefault(require("./StatusExpand"));
var _UserExpand = _interopRequireDefault(require("./UserExpand"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The MilestoneExpands model module.
 * @module model/MilestoneExpands
 * @version 2.0.0
 */
var MilestoneExpands = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MilestoneExpands</code>.
   * @alias module:model/MilestoneExpands
   */
  function MilestoneExpands() {
    _classCallCheck(this, MilestoneExpands);
    MilestoneExpands.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(MilestoneExpands, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>MilestoneExpands</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MilestoneExpands} obj Optional instance to populate.
     * @return {module:model/MilestoneExpands} The populated <code>MilestoneExpands</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MilestoneExpands();
        if (data.hasOwnProperty('issues')) {
          obj['issues'] = _ApiClient["default"].convertToType(data['issues'], [_IssueExpand["default"]]);
        }
        if (data.hasOwnProperty('milestone_stats')) {
          obj['milestone_stats'] = _ApiClient["default"].convertToType(data['milestone_stats'], [_MilestoneStatsExpand["default"]]);
        }
        if (data.hasOwnProperty('milestone_types')) {
          obj['milestone_types'] = _ApiClient["default"].convertToType(data['milestone_types'], [_MilestoneTypeExpand["default"]]);
        }
        if (data.hasOwnProperty('milestones')) {
          obj['milestones'] = _ApiClient["default"].convertToType(data['milestones'], [_MilestoneExpand["default"]]);
        }
        if (data.hasOwnProperty('statuses')) {
          obj['statuses'] = _ApiClient["default"].convertToType(data['statuses'], [_StatusExpand["default"]]);
        }
        if (data.hasOwnProperty('users')) {
          obj['users'] = _ApiClient["default"].convertToType(data['users'], [_UserExpand["default"]]);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>MilestoneExpands</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>MilestoneExpands</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      if (data['issues']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['issues'])) {
          throw new Error("Expected the field `issues` to be an array in the JSON data but got " + data['issues']);
        }
        // validate the optional field `issues` (array)
        var _iterator = _createForOfIteratorHelper(data['issues']),
          _step;
        try {
          for (_iterator.s(); !(_step = _iterator.n()).done;) {
            var item = _step.value;
            _IssueExpand["default"].validateJsonObject(item);
          }
        } catch (err) {
          _iterator.e(err);
        } finally {
          _iterator.f();
        }
        ;
      }
      if (data['milestone_stats']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['milestone_stats'])) {
          throw new Error("Expected the field `milestone_stats` to be an array in the JSON data but got " + data['milestone_stats']);
        }
        // validate the optional field `milestone_stats` (array)
        var _iterator2 = _createForOfIteratorHelper(data['milestone_stats']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var _item = _step2.value;
            _MilestoneStatsExpand["default"].validateJsonObject(_item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      if (data['milestone_types']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['milestone_types'])) {
          throw new Error("Expected the field `milestone_types` to be an array in the JSON data but got " + data['milestone_types']);
        }
        // validate the optional field `milestone_types` (array)
        var _iterator3 = _createForOfIteratorHelper(data['milestone_types']),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item2 = _step3.value;
            _MilestoneTypeExpand["default"].validateJsonObject(_item2);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
        ;
      }
      if (data['milestones']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['milestones'])) {
          throw new Error("Expected the field `milestones` to be an array in the JSON data but got " + data['milestones']);
        }
        // validate the optional field `milestones` (array)
        var _iterator4 = _createForOfIteratorHelper(data['milestones']),
          _step4;
        try {
          for (_iterator4.s(); !(_step4 = _iterator4.n()).done;) {
            var _item3 = _step4.value;
            _MilestoneExpand["default"].validateJsonObject(_item3);
          }
        } catch (err) {
          _iterator4.e(err);
        } finally {
          _iterator4.f();
        }
        ;
      }
      if (data['statuses']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['statuses'])) {
          throw new Error("Expected the field `statuses` to be an array in the JSON data but got " + data['statuses']);
        }
        // validate the optional field `statuses` (array)
        var _iterator5 = _createForOfIteratorHelper(data['statuses']),
          _step5;
        try {
          for (_iterator5.s(); !(_step5 = _iterator5.n()).done;) {
            var _item4 = _step5.value;
            _StatusExpand["default"].validateJsonObject(_item4);
          }
        } catch (err) {
          _iterator5.e(err);
        } finally {
          _iterator5.f();
        }
        ;
      }
      if (data['users']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['users'])) {
          throw new Error("Expected the field `users` to be an array in the JSON data but got " + data['users']);
        }
        // validate the optional field `users` (array)
        var _iterator6 = _createForOfIteratorHelper(data['users']),
          _step6;
        try {
          for (_iterator6.s(); !(_step6 = _iterator6.n()).done;) {
            var _item5 = _step6.value;
            _UserExpand["default"].validateJsonObject(_item5);
          }
        } catch (err) {
          _iterator6.e(err);
        } finally {
          _iterator6.f();
        }
        ;
      }
      return true;
    }
  }]);
  return MilestoneExpands;
}();
/**
 * @member {Array.<module:model/IssueExpand>} issues
 */
MilestoneExpands.prototype['issues'] = undefined;

/**
 * @member {Array.<module:model/MilestoneStatsExpand>} milestone_stats
 */
MilestoneExpands.prototype['milestone_stats'] = undefined;

/**
 * @member {Array.<module:model/MilestoneTypeExpand>} milestone_types
 */
MilestoneExpands.prototype['milestone_types'] = undefined;

/**
 * @member {Array.<module:model/MilestoneExpand>} milestones
 */
MilestoneExpands.prototype['milestones'] = undefined;

/**
 * @member {Array.<module:model/StatusExpand>} statuses
 */
MilestoneExpands.prototype['statuses'] = undefined;

/**
 * @member {Array.<module:model/UserExpand>} users
 */
MilestoneExpands.prototype['users'] = undefined;
var _default = MilestoneExpands;
exports["default"] = _default;
