// @testmo/testmo-api/dist/model/Project.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Project model module.
 * @module model/Project
 * @version 2.0.0
 */
var Project = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Project</code>.
   * @alias module:model/Project
   */
  function Project() {
    _classCallCheck(this, Project);
    Project.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Project, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Project</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Project} obj Optional instance to populate.
     * @return {module:model/Project} The populated <code>Project</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Project();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('is_completed')) {
          obj['is_completed'] = _ApiClient["default"].convertToType(data['is_completed'], 'Boolean');
        }
        if (data.hasOwnProperty('milestone_count')) {
          obj['milestone_count'] = _ApiClient["default"].convertToType(data['milestone_count'], 'Number');
        }
        if (data.hasOwnProperty('milestone_active_count')) {
          obj['milestone_active_count'] = _ApiClient["default"].convertToType(data['milestone_active_count'], 'Number');
        }
        if (data.hasOwnProperty('milestone_completed_count')) {
          obj['milestone_completed_count'] = _ApiClient["default"].convertToType(data['milestone_completed_count'], 'Number');
        }
        if (data.hasOwnProperty('run_count')) {
          obj['run_count'] = _ApiClient["default"].convertToType(data['run_count'], 'Number');
        }
        if (data.hasOwnProperty('run_active_count')) {
          obj['run_active_count'] = _ApiClient["default"].convertToType(data['run_active_count'], 'Number');
        }
        if (data.hasOwnProperty('run_closed_count')) {
          obj['run_closed_count'] = _ApiClient["default"].convertToType(data['run_closed_count'], 'Number');
        }
        if (data.hasOwnProperty('session_count')) {
          obj['session_count'] = _ApiClient["default"].convertToType(data['session_count'], 'Number');
        }
        if (data.hasOwnProperty('session_active_count')) {
          obj['session_active_count'] = _ApiClient["default"].convertToType(data['session_active_count'], 'Number');
        }
        if (data.hasOwnProperty('session_closed_count')) {
          obj['session_closed_count'] = _ApiClient["default"].convertToType(data['session_closed_count'], 'Number');
        }
        if (data.hasOwnProperty('automation_source_count')) {
          obj['automation_source_count'] = _ApiClient["default"].convertToType(data['automation_source_count'], 'Number');
        }
        if (data.hasOwnProperty('automation_source_active_count')) {
          obj['automation_source_active_count'] = _ApiClient["default"].convertToType(data['automation_source_active_count'], 'Number');
        }
        if (data.hasOwnProperty('automation_source_retired_count')) {
          obj['automation_source_retired_count'] = _ApiClient["default"].convertToType(data['automation_source_retired_count'], 'Number');
        }
        if (data.hasOwnProperty('automation_run_count')) {
          obj['automation_run_count'] = _ApiClient["default"].convertToType(data['automation_run_count'], 'Number');
        }
        if (data.hasOwnProperty('automation_run_active_count')) {
          obj['automation_run_active_count'] = _ApiClient["default"].convertToType(data['automation_run_active_count'], 'Number');
        }
        if (data.hasOwnProperty('automation_run_completed_count')) {
          obj['automation_run_completed_count'] = _ApiClient["default"].convertToType(data['automation_run_completed_count'], 'Number');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('created_by')) {
          obj['created_by'] = _ApiClient["default"].convertToType(data['created_by'], 'Number');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_by')) {
          obj['updated_by'] = _ApiClient["default"].convertToType(data['updated_by'], 'Number');
        }
        if (data.hasOwnProperty('completed_at')) {
          obj['completed_at'] = _ApiClient["default"].convertToType(data['completed_at'], 'Date');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Project</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Project</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      return true;
    }
  }]);
  return Project;
}();
/**
 * @member {Number} id
 */
Project.prototype['id'] = undefined;

/**
 * @member {String} name
 */
Project.prototype['name'] = undefined;

/**
 * @member {Boolean} is_completed
 */
Project.prototype['is_completed'] = undefined;

/**
 * @member {Number} milestone_count
 */
Project.prototype['milestone_count'] = undefined;

/**
 * @member {Number} milestone_active_count
 */
Project.prototype['milestone_active_count'] = undefined;

/**
 * @member {Number} milestone_completed_count
 */
Project.prototype['milestone_completed_count'] = undefined;

/**
 * @member {Number} run_count
 */
Project.prototype['run_count'] = undefined;

/**
 * @member {Number} run_active_count
 */
Project.prototype['run_active_count'] = undefined;

/**
 * @member {Number} run_closed_count
 */
Project.prototype['run_closed_count'] = undefined;

/**
 * @member {Number} session_count
 */
Project.prototype['session_count'] = undefined;

/**
 * @member {Number} session_active_count
 */
Project.prototype['session_active_count'] = undefined;

/**
 * @member {Number} session_closed_count
 */
Project.prototype['session_closed_count'] = undefined;

/**
 * @member {Number} automation_source_count
 */
Project.prototype['automation_source_count'] = undefined;

/**
 * @member {Number} automation_source_active_count
 */
Project.prototype['automation_source_active_count'] = undefined;

/**
 * @member {Number} automation_source_retired_count
 */
Project.prototype['automation_source_retired_count'] = undefined;

/**
 * @member {Number} automation_run_count
 */
Project.prototype['automation_run_count'] = undefined;

/**
 * @member {Number} automation_run_active_count
 */
Project.prototype['automation_run_active_count'] = undefined;

/**
 * @member {Number} automation_run_completed_count
 */
Project.prototype['automation_run_completed_count'] = undefined;

/**
 * @member {Date} created_at
 */
Project.prototype['created_at'] = undefined;

/**
 * @member {Number} created_by
 */
Project.prototype['created_by'] = undefined;

/**
 * @member {Date} updated_at
 */
Project.prototype['updated_at'] = undefined;

/**
 * @member {Number} updated_by
 */
Project.prototype['updated_by'] = undefined;

/**
 * @member {Date} completed_at
 */
Project.prototype['completed_at'] = undefined;
var _default = Project;
exports["default"] = _default;
