// @testmo/testmo-api/dist/model/Session.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Session model module.
 * @module model/Session
 * @version 2.0.0
 */
var Session = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Session</code>.
   * @alias module:model/Session
   */
  function Session() {
    _classCallCheck(this, Session);
    Session.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Session, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Session</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Session} obj Optional instance to populate.
     * @return {module:model/Session} The populated <code>Session</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Session();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('project_id')) {
          obj['project_id'] = _ApiClient["default"].convertToType(data['project_id'], 'Number');
        }
        if (data.hasOwnProperty('template_id')) {
          obj['template_id'] = _ApiClient["default"].convertToType(data['template_id'], 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('config_id')) {
          obj['config_id'] = _ApiClient["default"].convertToType(data['config_id'], 'Number');
        }
        if (data.hasOwnProperty('milestone_id')) {
          obj['milestone_id'] = _ApiClient["default"].convertToType(data['milestone_id'], 'Number');
        }
        if (data.hasOwnProperty('state_id')) {
          obj['state_id'] = _ApiClient["default"].convertToType(data['state_id'], 'Number');
        }
        if (data.hasOwnProperty('assignee_id')) {
          obj['assignee_id'] = _ApiClient["default"].convertToType(data['assignee_id'], 'Number');
        }
        if (data.hasOwnProperty('estimate')) {
          obj['estimate'] = _ApiClient["default"].convertToType(data['estimate'], 'Number');
        }
        if (data.hasOwnProperty('forecast')) {
          obj['forecast'] = _ApiClient["default"].convertToType(data['forecast'], 'Number');
        }
        if (data.hasOwnProperty('elapsed')) {
          obj['elapsed'] = _ApiClient["default"].convertToType(data['elapsed'], 'Number');
        }
        if (data.hasOwnProperty('is_started')) {
          obj['is_started'] = _ApiClient["default"].convertToType(data['is_started'], 'Boolean');
        }
        if (data.hasOwnProperty('is_closed')) {
          obj['is_closed'] = _ApiClient["default"].convertToType(data['is_closed'], 'Boolean');
        }
        if (data.hasOwnProperty('issues')) {
          obj['issues'] = _ApiClient["default"].convertToType(data['issues'], ['Number']);
        }
        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }
        if (data.hasOwnProperty('untested_count')) {
          obj['untested_count'] = _ApiClient["default"].convertToType(data['untested_count'], 'Number');
        }
        if (data.hasOwnProperty('status1_count')) {
          obj['status1_count'] = _ApiClient["default"].convertToType(data['status1_count'], 'Number');
        }
        if (data.hasOwnProperty('status2_count')) {
          obj['status2_count'] = _ApiClient["default"].convertToType(data['status2_count'], 'Number');
        }
        if (data.hasOwnProperty('status3_count')) {
          obj['status3_count'] = _ApiClient["default"].convertToType(data['status3_count'], 'Number');
        }
        if (data.hasOwnProperty('status4_count')) {
          obj['status4_count'] = _ApiClient["default"].convertToType(data['status4_count'], 'Number');
        }
        if (data.hasOwnProperty('status5_count')) {
          obj['status5_count'] = _ApiClient["default"].convertToType(data['status5_count'], 'Number');
        }
        if (data.hasOwnProperty('status6_count')) {
          obj['status6_count'] = _ApiClient["default"].convertToType(data['status6_count'], 'Number');
        }
        if (data.hasOwnProperty('status7_count')) {
          obj['status7_count'] = _ApiClient["default"].convertToType(data['status7_count'], 'Number');
        }
        if (data.hasOwnProperty('status8_count')) {
          obj['status8_count'] = _ApiClient["default"].convertToType(data['status8_count'], 'Number');
        }
        if (data.hasOwnProperty('status9_count')) {
          obj['status9_count'] = _ApiClient["default"].convertToType(data['status9_count'], 'Number');
        }
        if (data.hasOwnProperty('status10_count')) {
          obj['status10_count'] = _ApiClient["default"].convertToType(data['status10_count'], 'Number');
        }
        if (data.hasOwnProperty('status11_count')) {
          obj['status11_count'] = _ApiClient["default"].convertToType(data['status11_count'], 'Number');
        }
        if (data.hasOwnProperty('status12_count')) {
          obj['status12_count'] = _ApiClient["default"].convertToType(data['status12_count'], 'Number');
        }
        if (data.hasOwnProperty('status13_count')) {
          obj['status13_count'] = _ApiClient["default"].convertToType(data['status13_count'], 'Number');
        }
        if (data.hasOwnProperty('status14_count')) {
          obj['status14_count'] = _ApiClient["default"].convertToType(data['status14_count'], 'Number');
        }
        if (data.hasOwnProperty('status15_count')) {
          obj['status15_count'] = _ApiClient["default"].convertToType(data['status15_count'], 'Number');
        }
        if (data.hasOwnProperty('status16_count')) {
          obj['status16_count'] = _ApiClient["default"].convertToType(data['status16_count'], 'Number');
        }
        if (data.hasOwnProperty('status17_count')) {
          obj['status17_count'] = _ApiClient["default"].convertToType(data['status17_count'], 'Number');
        }
        if (data.hasOwnProperty('status18_count')) {
          obj['status18_count'] = _ApiClient["default"].convertToType(data['status18_count'], 'Number');
        }
        if (data.hasOwnProperty('status19_count')) {
          obj['status19_count'] = _ApiClient["default"].convertToType(data['status19_count'], 'Number');
        }
        if (data.hasOwnProperty('status20_count')) {
          obj['status20_count'] = _ApiClient["default"].convertToType(data['status20_count'], 'Number');
        }
        if (data.hasOwnProperty('status21_count')) {
          obj['status21_count'] = _ApiClient["default"].convertToType(data['status21_count'], 'Number');
        }
        if (data.hasOwnProperty('status22_count')) {
          obj['status22_count'] = _ApiClient["default"].convertToType(data['status22_count'], 'Number');
        }
        if (data.hasOwnProperty('status23_count')) {
          obj['status23_count'] = _ApiClient["default"].convertToType(data['status23_count'], 'Number');
        }
        if (data.hasOwnProperty('status24_count')) {
          obj['status24_count'] = _ApiClient["default"].convertToType(data['status24_count'], 'Number');
        }
        if (data.hasOwnProperty('success_count')) {
          obj['success_count'] = _ApiClient["default"].convertToType(data['success_count'], 'Number');
        }
        if (data.hasOwnProperty('failure_count')) {
          obj['failure_count'] = _ApiClient["default"].convertToType(data['failure_count'], 'Number');
        }
        if (data.hasOwnProperty('completed_count')) {
          obj['completed_count'] = _ApiClient["default"].convertToType(data['completed_count'], 'Number');
        }
        if (data.hasOwnProperty('total_count')) {
          obj['total_count'] = _ApiClient["default"].convertToType(data['total_count'], 'Number');
        }
        if (data.hasOwnProperty('started_at')) {
          obj['started_at'] = _ApiClient["default"].convertToType(data['started_at'], 'Date');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('created_by')) {
          obj['created_by'] = _ApiClient["default"].convertToType(data['created_by'], 'Number');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_by')) {
          obj['updated_by'] = _ApiClient["default"].convertToType(data['updated_by'], 'Number');
        }
        if (data.hasOwnProperty('closed_at')) {
          obj['closed_at'] = _ApiClient["default"].convertToType(data['closed_at'], 'Date');
        }
        if (data.hasOwnProperty('closed_by')) {
          obj['closed_by'] = _ApiClient["default"].convertToType(data['closed_by'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Session</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Session</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['issues'])) {
        throw new Error("Expected the field `issues` to be an array in the JSON data but got " + data['issues']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['tags'])) {
        throw new Error("Expected the field `tags` to be an array in the JSON data but got " + data['tags']);
      }
      return true;
    }
  }]);
  return Session;
}();
/**
 * @member {Number} id
 */
Session.prototype['id'] = undefined;

/**
 * @member {Number} project_id
 */
Session.prototype['project_id'] = undefined;

/**
 * @member {Number} template_id
 */
Session.prototype['template_id'] = undefined;

/**
 * @member {String} name
 */
Session.prototype['name'] = undefined;

/**
 * @member {Number} config_id
 */
Session.prototype['config_id'] = undefined;

/**
 * @member {Number} milestone_id
 */
Session.prototype['milestone_id'] = undefined;

/**
 * @member {Number} state_id
 */
Session.prototype['state_id'] = undefined;

/**
 * @member {Number} assignee_id
 */
Session.prototype['assignee_id'] = undefined;

/**
 * @member {Number} estimate
 */
Session.prototype['estimate'] = undefined;

/**
 * @member {Number} forecast
 */
Session.prototype['forecast'] = undefined;

/**
 * @member {Number} elapsed
 */
Session.prototype['elapsed'] = undefined;

/**
 * @member {Boolean} is_started
 */
Session.prototype['is_started'] = undefined;

/**
 * @member {Boolean} is_closed
 */
Session.prototype['is_closed'] = undefined;

/**
 * @member {Array.<Number>} issues
 */
Session.prototype['issues'] = undefined;

/**
 * @member {Array.<String>} tags
 */
Session.prototype['tags'] = undefined;

/**
 * @member {Number} untested_count
 */
Session.prototype['untested_count'] = undefined;

/**
 * @member {Number} status1_count
 */
Session.prototype['status1_count'] = undefined;

/**
 * @member {Number} status2_count
 */
Session.prototype['status2_count'] = undefined;

/**
 * @member {Number} status3_count
 */
Session.prototype['status3_count'] = undefined;

/**
 * @member {Number} status4_count
 */
Session.prototype['status4_count'] = undefined;

/**
 * @member {Number} status5_count
 */
Session.prototype['status5_count'] = undefined;

/**
 * @member {Number} status6_count
 */
Session.prototype['status6_count'] = undefined;

/**
 * @member {Number} status7_count
 */
Session.prototype['status7_count'] = undefined;

/**
 * @member {Number} status8_count
 */
Session.prototype['status8_count'] = undefined;

/**
 * @member {Number} status9_count
 */
Session.prototype['status9_count'] = undefined;

/**
 * @member {Number} status10_count
 */
Session.prototype['status10_count'] = undefined;

/**
 * @member {Number} status11_count
 */
Session.prototype['status11_count'] = undefined;

/**
 * @member {Number} status12_count
 */
Session.prototype['status12_count'] = undefined;

/**
 * @member {Number} status13_count
 */
Session.prototype['status13_count'] = undefined;

/**
 * @member {Number} status14_count
 */
Session.prototype['status14_count'] = undefined;

/**
 * @member {Number} status15_count
 */
Session.prototype['status15_count'] = undefined;

/**
 * @member {Number} status16_count
 */
Session.prototype['status16_count'] = undefined;

/**
 * @member {Number} status17_count
 */
Session.prototype['status17_count'] = undefined;

/**
 * @member {Number} status18_count
 */
Session.prototype['status18_count'] = undefined;

/**
 * @member {Number} status19_count
 */
Session.prototype['status19_count'] = undefined;

/**
 * @member {Number} status20_count
 */
Session.prototype['status20_count'] = undefined;

/**
 * @member {Number} status21_count
 */
Session.prototype['status21_count'] = undefined;

/**
 * @member {Number} status22_count
 */
Session.prototype['status22_count'] = undefined;

/**
 * @member {Number} status23_count
 */
Session.prototype['status23_count'] = undefined;

/**
 * @member {Number} status24_count
 */
Session.prototype['status24_count'] = undefined;

/**
 * @member {Number} success_count
 */
Session.prototype['success_count'] = undefined;

/**
 * @member {Number} failure_count
 */
Session.prototype['failure_count'] = undefined;

/**
 * @member {Number} completed_count
 */
Session.prototype['completed_count'] = undefined;

/**
 * @member {Number} total_count
 */
Session.prototype['total_count'] = undefined;

/**
 * @member {Date} started_at
 */
Session.prototype['started_at'] = undefined;

/**
 * @member {Date} created_at
 */
Session.prototype['created_at'] = undefined;

/**
 * @member {Number} created_by
 */
Session.prototype['created_by'] = undefined;

/**
 * @member {Date} updated_at
 */
Session.prototype['updated_at'] = undefined;

/**
 * @member {Number} updated_by
 */
Session.prototype['updated_by'] = undefined;

/**
 * @member {Date} closed_at
 */
Session.prototype['closed_at'] = undefined;

/**
 * @member {Number} closed_by
 */
Session.prototype['closed_by'] = undefined;
var _default = Session;
exports["default"] = _default;
