// @testmo/testmo-api/dist/model/StatusExpand.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The StatusExpand model module.
 * @module model/StatusExpand
 * @version 2.0.0
 */
var StatusExpand = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>StatusExpand</code>.
   * @alias module:model/StatusExpand
   */
  function StatusExpand() {
    _classCallCheck(this, StatusExpand);
    StatusExpand.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(StatusExpand, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>StatusExpand</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/StatusExpand} obj Optional instance to populate.
     * @return {module:model/StatusExpand} The populated <code>StatusExpand</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new StatusExpand();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('system_name')) {
          obj['system_name'] = _ApiClient["default"].convertToType(data['system_name'], 'String');
        }
        if (data.hasOwnProperty('color')) {
          obj['color'] = _ApiClient["default"].convertToType(data['color'], 'String');
        }
        if (data.hasOwnProperty('is_final')) {
          obj['is_final'] = _ApiClient["default"].convertToType(data['is_final'], 'Boolean');
        }
        if (data.hasOwnProperty('is_untested')) {
          obj['is_untested'] = _ApiClient["default"].convertToType(data['is_untested'], 'Boolean');
        }
        if (data.hasOwnProperty('is_passed')) {
          obj['is_passed'] = _ApiClient["default"].convertToType(data['is_passed'], 'Boolean');
        }
        if (data.hasOwnProperty('is_failed')) {
          obj['is_failed'] = _ApiClient["default"].convertToType(data['is_failed'], 'Boolean');
        }
        if (data.hasOwnProperty('aliases')) {
          obj['aliases'] = _ApiClient["default"].convertToType(data['aliases'], ['String']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>StatusExpand</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>StatusExpand</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['system_name'] && !(typeof data['system_name'] === 'string' || data['system_name'] instanceof String)) {
        throw new Error("Expected the field `system_name` to be a primitive type in the JSON string but got " + data['system_name']);
      }
      // ensure the json data is a string
      if (data['color'] && !(typeof data['color'] === 'string' || data['color'] instanceof String)) {
        throw new Error("Expected the field `color` to be a primitive type in the JSON string but got " + data['color']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['aliases'])) {
        throw new Error("Expected the field `aliases` to be an array in the JSON data but got " + data['aliases']);
      }
      return true;
    }
  }]);
  return StatusExpand;
}();
/**
 * @member {Number} id
 */
StatusExpand.prototype['id'] = undefined;

/**
 * @member {String} name
 */
StatusExpand.prototype['name'] = undefined;

/**
 * @member {String} system_name
 */
StatusExpand.prototype['system_name'] = undefined;

/**
 * @member {String} color
 */
StatusExpand.prototype['color'] = undefined;

/**
 * @member {Boolean} is_final
 */
StatusExpand.prototype['is_final'] = undefined;

/**
 * @member {Boolean} is_untested
 */
StatusExpand.prototype['is_untested'] = undefined;

/**
 * @member {Boolean} is_passed
 */
StatusExpand.prototype['is_passed'] = undefined;

/**
 * @member {Boolean} is_failed
 */
StatusExpand.prototype['is_failed'] = undefined;

/**
 * @member {Array.<String>} aliases
 */
StatusExpand.prototype['aliases'] = undefined;
var _default = StatusExpand;
exports["default"] = _default;
