// @testmo/testmo-api/dist/model/User.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The User model module.
 * @module model/User
 * @version 2.0.0
 */
var User = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>User</code>.
   * @alias module:model/User
   */
  function User() {
    _classCallCheck(this, User);
    User.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(User, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>User</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/User} obj Optional instance to populate.
     * @return {module:model/User} The populated <code>User</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new User();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'Number');
        }
        if (data.hasOwnProperty('timezone')) {
          obj['timezone'] = _ApiClient["default"].convertToType(data['timezone'], 'String');
        }
        if (data.hasOwnProperty('date_format')) {
          obj['date_format'] = _ApiClient["default"].convertToType(data['date_format'], 'String');
        }
        if (data.hasOwnProperty('time_format')) {
          obj['time_format'] = _ApiClient["default"].convertToType(data['time_format'], 'String');
        }
        if (data.hasOwnProperty('notifications')) {
          obj['notifications'] = _ApiClient["default"].convertToType(data['notifications'], 'Number');
        }
        if (data.hasOwnProperty('tfa_enabled')) {
          obj['tfa_enabled'] = _ApiClient["default"].convertToType(data['tfa_enabled'], 'Boolean');
        }
        if (data.hasOwnProperty('is_active')) {
          obj['is_active'] = _ApiClient["default"].convertToType(data['is_active'], 'Boolean');
        }
        if (data.hasOwnProperty('is_forgotten')) {
          obj['is_forgotten'] = _ApiClient["default"].convertToType(data['is_forgotten'], 'Boolean');
        }
        if (data.hasOwnProperty('is_api')) {
          obj['is_api'] = _ApiClient["default"].convertToType(data['is_api'], 'Boolean');
        }
        if (data.hasOwnProperty('role_id')) {
          obj['role_id'] = _ApiClient["default"].convertToType(data['role_id'], 'Number');
        }
        if (data.hasOwnProperty('groups')) {
          obj['groups'] = _ApiClient["default"].convertToType(data['groups'], ['Number']);
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('created_by')) {
          obj['created_by'] = _ApiClient["default"].convertToType(data['created_by'], 'Number');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_by')) {
          obj['updated_by'] = _ApiClient["default"].convertToType(data['updated_by'], 'Number');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>User</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>User</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['email'] && !(typeof data['email'] === 'string' || data['email'] instanceof String)) {
        throw new Error("Expected the field `email` to be a primitive type in the JSON string but got " + data['email']);
      }
      // ensure the json data is a string
      if (data['timezone'] && !(typeof data['timezone'] === 'string' || data['timezone'] instanceof String)) {
        throw new Error("Expected the field `timezone` to be a primitive type in the JSON string but got " + data['timezone']);
      }
      // ensure the json data is a string
      if (data['date_format'] && !(typeof data['date_format'] === 'string' || data['date_format'] instanceof String)) {
        throw new Error("Expected the field `date_format` to be a primitive type in the JSON string but got " + data['date_format']);
      }
      // ensure the json data is a string
      if (data['time_format'] && !(typeof data['time_format'] === 'string' || data['time_format'] instanceof String)) {
        throw new Error("Expected the field `time_format` to be a primitive type in the JSON string but got " + data['time_format']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['groups'])) {
        throw new Error("Expected the field `groups` to be an array in the JSON data but got " + data['groups']);
      }
      return true;
    }
  }]);
  return User;
}();
/**
 * @member {Number} id
 */
User.prototype['id'] = undefined;

/**
 * @member {String} name
 */
User.prototype['name'] = undefined;

/**
 * @member {String} email
 */
User.prototype['email'] = undefined;

/**
 * @member {Number} type
 */
User.prototype['type'] = undefined;

/**
 * @member {String} timezone
 */
User.prototype['timezone'] = undefined;

/**
 * @member {String} date_format
 */
User.prototype['date_format'] = undefined;

/**
 * @member {String} time_format
 */
User.prototype['time_format'] = undefined;

/**
 * @member {module:model/User.NotificationsEnum} notifications
 */
User.prototype['notifications'] = undefined;

/**
 * @member {Boolean} tfa_enabled
 */
User.prototype['tfa_enabled'] = undefined;

/**
 * @member {Boolean} is_active
 */
User.prototype['is_active'] = undefined;

/**
 * @member {Boolean} is_forgotten
 */
User.prototype['is_forgotten'] = undefined;

/**
 * @member {Boolean} is_api
 */
User.prototype['is_api'] = undefined;

/**
 * @member {Number} role_id
 */
User.prototype['role_id'] = undefined;

/**
 * @member {Array.<Number>} groups
 */
User.prototype['groups'] = undefined;

/**
 * @member {Date} created_at
 */
User.prototype['created_at'] = undefined;

/**
 * @member {Number} created_by
 */
User.prototype['created_by'] = undefined;

/**
 * @member {Date} updated_at
 */
User.prototype['updated_at'] = undefined;

/**
 * @member {Number} updated_by
 */
User.prototype['updated_by'] = undefined;

/**
 * Allowed values for the <code>notifications</code> property.
 * @enum {Number}
 * @readonly
 */
User['NotificationsEnum'] = {
  /**
   * value: 1
   * @const
   */
  "1": 1,
  /**
   * value: 2
   * @const
   */
  "2": 2,
  /**
   * value: 3
   * @const
   */
  "3": 3
};
var _default = User;
exports["default"] = _default;
