// @testmo/testmo-api/dist/api/AutomationSourcesApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AutomationSourcePage = _interopRequireDefault(require("../model/AutomationSourcePage"));
var _AutomationSourceRow = _interopRequireDefault(require("../model/AutomationSourceRow"));
var _ErrorCollection = _interopRequireDefault(require("../model/ErrorCollection"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
* AutomationSources service.
* @module api/AutomationSourcesApi
* @version 2.0.0
*/
var AutomationSourcesApi = /*#__PURE__*/function () {
  /**
  * Constructs a new AutomationSourcesApi. 
  * @alias module:api/AutomationSourcesApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function AutomationSourcesApi(apiClient) {
    _classCallCheck(this, AutomationSourcesApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Get a single automation source
   * Returns a single automation source.
   * @param {Number} automationSourceId ID of the automation source to return.
   * @param {Object} opts Optional parameters
   * @param {String} opts.expands Comma-separated list of expands to return.
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/AutomationSourceRow} and HTTP response
   */
  _createClass(AutomationSourcesApi, [{
    key: "getAutomationSourceWithHttpInfo",
    value: function getAutomationSourceWithHttpInfo(automationSourceId, opts) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'automationSourceId' is set
      if (automationSourceId === undefined || automationSourceId === null) {
        throw new Error("Missing the required parameter 'automationSourceId' when calling getAutomationSource");
      }
      var pathParams = {
        'automation_source_id': automationSourceId
      };
      var queryParams = {
        'expands': opts['expands']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AutomationSourceRow["default"];
      return this.apiClient.callApi('/api/v1/automation/sources/{automation_source_id}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get a single automation source
     * Returns a single automation source.
     * @param {Number} automationSourceId ID of the automation source to return.
     * @param {Object} opts Optional parameters
     * @param {String} opts.expands Comma-separated list of expands to return.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/AutomationSourceRow}
     */
  }, {
    key: "getAutomationSource",
    value: function getAutomationSource(automationSourceId, opts) {
      return this.getAutomationSourceWithHttpInfo(automationSourceId, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get all automation sources for a project
     * Returns a page of automation sources for a project.
     * @param {Number} projectId ID of the project.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.page Number of page to return.
     * @param {module:model/Number} opts.perPage Maximum number of items to return per page.
     * @param {String} opts.expands Comma-separated list of expands to return.
     * @param {module:model/String} opts.sort Sort field for the list of automation sources.
     * @param {module:model/String} opts.order Sort order (ascending or descending).
     * @param {Boolean} opts.isRetired Limit result to active or retired automation sources only.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/AutomationSourcePage} and HTTP response
     */
  }, {
    key: "getAutomationSourcePageWithHttpInfo",
    value: function getAutomationSourcePageWithHttpInfo(projectId, opts) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'projectId' is set
      if (projectId === undefined || projectId === null) {
        throw new Error("Missing the required parameter 'projectId' when calling getAutomationSourcePage");
      }
      var pathParams = {
        'project_id': projectId
      };
      var queryParams = {
        'page': opts['page'],
        'per_page': opts['perPage'],
        'expands': opts['expands'],
        'sort': opts['sort'],
        'order': opts['order'],
        'is_retired': opts['isRetired']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _AutomationSourcePage["default"];
      return this.apiClient.callApi('/api/v1/projects/{project_id}/automation/sources', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get all automation sources for a project
     * Returns a page of automation sources for a project.
     * @param {Number} projectId ID of the project.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.page Number of page to return.
     * @param {module:model/Number} opts.perPage Maximum number of items to return per page.
     * @param {String} opts.expands Comma-separated list of expands to return.
     * @param {module:model/String} opts.sort Sort field for the list of automation sources.
     * @param {module:model/String} opts.order Sort order (ascending or descending).
     * @param {Boolean} opts.isRetired Limit result to active or retired automation sources only.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/AutomationSourcePage}
     */
  }, {
    key: "getAutomationSourcePage",
    value: function getAutomationSourcePage(projectId, opts) {
      return this.getAutomationSourcePageWithHttpInfo(projectId, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
  return AutomationSourcesApi;
}();
exports["default"] = AutomationSourcesApi;
