// @testmo/testmo-api/dist/api/RunsApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ErrorCollection = _interopRequireDefault(require("../model/ErrorCollection"));
var _RunPage = _interopRequireDefault(require("../model/RunPage"));
var _RunRow = _interopRequireDefault(require("../model/RunRow"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
* Runs service.
* @module api/RunsApi
* @version 2.0.0
*/
var RunsApi = /*#__PURE__*/function () {
  /**
  * Constructs a new RunsApi. 
  * @alias module:api/RunsApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function RunsApi(apiClient) {
    _classCallCheck(this, RunsApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Get a single run
   * Returns a single run.
   * @param {Number} runId ID of the run to return.
   * @param {Object} opts Optional parameters
   * @param {String} opts.expands Comma-separated list of expands to return.
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/RunRow} and HTTP response
   */
  _createClass(RunsApi, [{
    key: "getRunWithHttpInfo",
    value: function getRunWithHttpInfo(runId, opts) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'runId' is set
      if (runId === undefined || runId === null) {
        throw new Error("Missing the required parameter 'runId' when calling getRun");
      }
      var pathParams = {
        'run_id': runId
      };
      var queryParams = {
        'expands': opts['expands']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _RunRow["default"];
      return this.apiClient.callApi('/api/v1/runs/{run_id}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get a single run
     * Returns a single run.
     * @param {Number} runId ID of the run to return.
     * @param {Object} opts Optional parameters
     * @param {String} opts.expands Comma-separated list of expands to return.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/RunRow}
     */
  }, {
    key: "getRun",
    value: function getRun(runId, opts) {
      return this.getRunWithHttpInfo(runId, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get all runs for a project
     * Returns a page of runs for a project.
     * @param {Number} projectId ID of the project.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.page Number of page to return.
     * @param {module:model/Number} opts.perPage Maximum number of items to return per page.
     * @param {String} opts.expands Comma-separated list of expands to return.
     * @param {module:model/String} opts.sort Sort field for the list of runs.
     * @param {module:model/String} opts.order Sort order (ascending or descending).
     * @param {Date} opts.closedAfter Limit result to runs closed after (in ISO8601 format and UTC time zone).
     * @param {Date} opts.closedBefore Limit result to runs closed before (in ISO8601 format and UTC time zone).
     * @param {String} opts.configId Comma-separated list of configurations to filter by.
     * @param {Date} opts.createdAfter Limit result to runs created after (in ISO8601 format and UTC time zone).
     * @param {Date} opts.createdBefore Limit result to runs created before (in ISO8601 format and UTC time zone).
     * @param {String} opts.createdBy Comma-separated list of users to filter by.
     * @param {Boolean} opts.isClosed Limit result to active or closed runs only.
     * @param {String} opts.milestoneId Comma-separated list of milestones to filter by.
     * @param {String} opts.stateId Comma-separated list of workflow states to filter by.
     * @param {String} opts.tags Comma-separated list of tags to filter by.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/RunPage} and HTTP response
     */
  }, {
    key: "getRunPageWithHttpInfo",
    value: function getRunPageWithHttpInfo(projectId, opts) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'projectId' is set
      if (projectId === undefined || projectId === null) {
        throw new Error("Missing the required parameter 'projectId' when calling getRunPage");
      }
      var pathParams = {
        'project_id': projectId
      };
      var queryParams = {
        'page': opts['page'],
        'per_page': opts['perPage'],
        'expands': opts['expands'],
        'sort': opts['sort'],
        'order': opts['order'],
        'closed_after': opts['closedAfter'],
        'closed_before': opts['closedBefore'],
        'config_id': opts['configId'],
        'created_after': opts['createdAfter'],
        'created_before': opts['createdBefore'],
        'created_by': opts['createdBy'],
        'is_closed': opts['isClosed'],
        'milestone_id': opts['milestoneId'],
        'state_id': opts['stateId'],
        'tags': opts['tags']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _RunPage["default"];
      return this.apiClient.callApi('/api/v1/projects/{project_id}/runs', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get all runs for a project
     * Returns a page of runs for a project.
     * @param {Number} projectId ID of the project.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.page Number of page to return.
     * @param {module:model/Number} opts.perPage Maximum number of items to return per page.
     * @param {String} opts.expands Comma-separated list of expands to return.
     * @param {module:model/String} opts.sort Sort field for the list of runs.
     * @param {module:model/String} opts.order Sort order (ascending or descending).
     * @param {Date} opts.closedAfter Limit result to runs closed after (in ISO8601 format and UTC time zone).
     * @param {Date} opts.closedBefore Limit result to runs closed before (in ISO8601 format and UTC time zone).
     * @param {String} opts.configId Comma-separated list of configurations to filter by.
     * @param {Date} opts.createdAfter Limit result to runs created after (in ISO8601 format and UTC time zone).
     * @param {Date} opts.createdBefore Limit result to runs created before (in ISO8601 format and UTC time zone).
     * @param {String} opts.createdBy Comma-separated list of users to filter by.
     * @param {Boolean} opts.isClosed Limit result to active or closed runs only.
     * @param {String} opts.milestoneId Comma-separated list of milestones to filter by.
     * @param {String} opts.stateId Comma-separated list of workflow states to filter by.
     * @param {String} opts.tags Comma-separated list of tags to filter by.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/RunPage}
     */
  }, {
    key: "getRunPage",
    value: function getRunPage(projectId, opts) {
      return this.getRunPageWithHttpInfo(projectId, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
  return RunsApi;
}();
exports["default"] = RunsApi;
