// @testmo/testmo-api/dist/api/UsersApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ErrorCollection = _interopRequireDefault(require("../model/ErrorCollection"));
var _ProjectUserPage = _interopRequireDefault(require("../model/ProjectUserPage"));
var _UserPage = _interopRequireDefault(require("../model/UserPage"));
var _UserRow = _interopRequireDefault(require("../model/UserRow"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Testmo REST API
                                                                                                                                                                                                                                                                                                                                                                                               * The Testmo REST API enables applications to interact with Testmo instances.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 2.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: contact@testmo.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
* Users service.
* @module api/UsersApi
* @version 2.0.0
*/
var UsersApi = /*#__PURE__*/function () {
  /**
  * Constructs a new UsersApi. 
  * @alias module:api/UsersApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function UsersApi(apiClient) {
    _classCallCheck(this, UsersApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Get a single user
   * Returns a single user.
   * @param {Number} userId ID of the user to return.
   * @param {Object} opts Optional parameters
   * @param {String} opts.expands Comma-separated list of expands to return.
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/UserRow} and HTTP response
   */
  _createClass(UsersApi, [{
    key: "getUserWithHttpInfo",
    value: function getUserWithHttpInfo(userId, opts) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'userId' is set
      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling getUser");
      }
      var pathParams = {
        'user_id': userId
      };
      var queryParams = {
        'expands': opts['expands']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _UserRow["default"];
      return this.apiClient.callApi('/api/v1/users/{user_id}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get a single user
     * Returns a single user.
     * @param {Number} userId ID of the user to return.
     * @param {Object} opts Optional parameters
     * @param {String} opts.expands Comma-separated list of expands to return.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/UserRow}
     */
  }, {
    key: "getUser",
    value: function getUser(userId, opts) {
      return this.getUserWithHttpInfo(userId, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get all users
     * Returns a page of users.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.page Number of page to return.
     * @param {module:model/Number} opts.perPage Maximum number of items to return per page.
     * @param {String} opts.expands Comma-separated list of expands to return.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/UserPage} and HTTP response
     */
  }, {
    key: "getUserPageWithHttpInfo",
    value: function getUserPageWithHttpInfo(opts) {
      opts = opts || {};
      var postBody = null;
      var pathParams = {};
      var queryParams = {
        'page': opts['page'],
        'per_page': opts['perPage'],
        'expands': opts['expands']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _UserPage["default"];
      return this.apiClient.callApi('/api/v1/users', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get all users
     * Returns a page of users.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.page Number of page to return.
     * @param {module:model/Number} opts.perPage Maximum number of items to return per page.
     * @param {String} opts.expands Comma-separated list of expands to return.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/UserPage}
     */
  }, {
    key: "getUserPage",
    value: function getUserPage(opts) {
      return this.getUserPageWithHttpInfo(opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }

    /**
     * Get all users for a project
     * Returns a page of users for a project.
     * @param {Number} projectId ID of the project.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.page Number of page to return.
     * @param {module:model/Number} opts.perPage Maximum number of items to return per page.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/ProjectUserPage} and HTTP response
     */
  }, {
    key: "getUserPageForProjectWithHttpInfo",
    value: function getUserPageForProjectWithHttpInfo(projectId, opts) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'projectId' is set
      if (projectId === undefined || projectId === null) {
        throw new Error("Missing the required parameter 'projectId' when calling getUserPageForProject");
      }
      var pathParams = {
        'project_id': projectId
      };
      var queryParams = {
        'page': opts['page'],
        'per_page': opts['perPage']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['bearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ProjectUserPage["default"];
      return this.apiClient.callApi('/api/v1/projects/{project_id}/users', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }

    /**
     * Get all users for a project
     * Returns a page of users for a project.
     * @param {Number} projectId ID of the project.
     * @param {Object} opts Optional parameters
     * @param {Number} opts.page Number of page to return.
     * @param {module:model/Number} opts.perPage Maximum number of items to return per page.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/ProjectUserPage}
     */
  }, {
    key: "getUserPageForProject",
    value: function getUserPageForProject(projectId, opts) {
      return this.getUserPageForProjectWithHttpInfo(projectId, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);
  return UsersApi;
}();
exports["default"] = UsersApi;
